vllm/core/block_manager_v2.py

```diff
@@ -467,9 +467,24 @@ def _can_swap(self,
             AllocStatus: The AllocStatus for swapping in/out the given 
                 sequence_group on to the 'device'.
         """
-        blocks = self._get_blocks_for_swap(seq_group, status)
+        # First determine the number of blocks that will be touched to swap the 
+        # already full blocks.
+        blocks = []
+        for seq in seq_group.get_seqs(status=status):
+            block_table = self.block_tables[seq.seq_id]
+            if block_table.blocks is not None:
+                for block in block_table.blocks:
+                    if block.is_full:
+                       blocks.append(block)
         num_blocks_touched = self.block_allocator.get_num_blocks_touched(
-            blocks, device, num_lookahead_slots)
+            blocks, device, num_lookahead_slots = 0)
+        # Next determine the number of blocks that will be touched to swap the 
+        # the partially full appendable blocks.
+        for seq in seq_group.get_seqs(status=status):
+            block_table = self.block_tables[seq.seq_id]
+            num_blocks_touched += block_table.get_num_blocks_touched_by_append_slots(
+                block_table.get_unseen_token_ids(seq.get_token_ids()),
+                num_lookahead_slots=num_lookahead_slots)
```

Comment:
Merge into `get_num_blocks_touched`

Reply:
Moved to get_num_blocks_touched. However I had to change the function signatures because I need to pass the mapping from the sequences to the number of unseen tokens and also the map of sequences to existing blocks for those sequence. The map is needed because we need to map the number of unseen tokens to the existing block structure of the corresponding sequence.